java/api/src/main/java/io/ray/api/Ray.java

```diff
@@ -74,7 +74,32 @@ public static boolean isInitialized() {
    * @return A ObjectRef instance that represents the in-store object.
    */
   public static <T> ObjectRef<T> put(T obj, BaseActorHandle owner) {
-    return internal().put(obj, owner);
+    return internal().put(obj, owner, false);
+  }
+
+  /**
+   * Store Arrow-serialized data in the object store.
+   *
+   * @param obj The Java object to be stored.
+   * @return A ObjectRef instance that represents the in-store object.
+   */
+  public static <T> ObjectRef<T> putArrow(T data) {
```

Comment:
I don't like this API design. Ideally, it shouldn't need any API changes. I believe the use case this PR suggests is:

1. The user serializes arrow data to byte[] or ByteBuffer.
2. The user invokes `Ray.putArrow(...)` to write serialized data to the object store with metadata "ARROW" and data as is.

I believe this is much better:

1. The user call `Ray.put(...)` with unserialized Arrow data objects.
2. Ray detects that the object to be serialized is an arrow data object, and invokes an arrow-specific method to serialize it to arrow IPC format represented by byte[] or ByteBuffer.
3. Ray writes serialized data to the object store with metadata "ARROW" and serialized data.

In theory, Ray Java should be able to support other IPC formats besides arrow. e.g. flatbuffer. I don't think adding new APIs is the right way to go.

Reply:
I think one of the key point is that the ray java worker lack of a universal serialization framework abstraction, which leads to that we need to do special handling for every type we want to support cross-language. It would be better that we have a serialization abstraction and the user can register custom serializer. In this way, there won't be no need for any api changes.

Reply:
Actually, I implemented a cross-language native serialization library before with out-of-band serialization and arrow zero-copy support. I can post the design doc here to introduce some new inputs if it's needed.

Reply:
>  I can post the design doc here to introduce some new inputs if it's needed.
That would be great!

Reply:
https://docs.google.com/document/d/1nrKrXnyRqiIQqLV1P6i3t6TXQEwduoyLeHLMT2DV-fc This is our serialization library for ray design doc @kira-lin

Reply:
I agree with @kfstorm that this should be supported in the existing serialization framework (as a msgpack extension?) without needing any separate API changes / any API changes, similar to how it works transparently in Python.

Reply:
Hi @chaokunyang 
sorry for late reply. I see it's a powerful serialization framework, and look forward to its integration. Just to make sure, it supports cross-language Arrow (de)serialization, and is zero-copy, right?

Reply:
Agreed, I was also considering task/object-level serialization metadata for the Rust API, if not possible, then at least cluster-level serialization toggling.

Reply:
> Hi @chaokunyang sorry for late reply. I see it's a powerful serialization framework, and look forward to its integration. Just to make sure, it supports cross-language Arrow (de)serialization, and is zero-copy, right?

@kira-lin Yes, it supports cross-language Arrow (de)serialization, and is zero-copy. For java serialization, we imeplements a MockWritableByteChannel whcih make us get ArrowRecordBatch/Table in advance, then allocate memory from plasma and then wrap it into DrirectBufferWritableByteChannel implememted by fury. So the whole process is zero-copy.